product/views.py

```diff
@@ -0,0 +1,43 @@
+import json
+from random                     import uniform
+
+from django.views               import View
+from django.http                import JsonResponse
+from django.core.exceptions     import ValidationError
+
+from product.models             import Product
+
+class ProductDetailView(View):
+    def get(self ,request):
+        try:
+            pk = request.GET.get('id',None)
+
+            if pk is not None:
+                product_list = Product.objects.filter(id=pk).values()
+                product      = Product.objects.get(id=pk)
+                descriptions = product.description.values()
+                images_url   = [url['url'] for url in product.image.values('url')]
+
+                result = [
+                    {
+                    'id'            : product.id,
+                    'korean_name'   : product.korean_name,
+                    'english_name'  : product.english_name,
+                    'price'         : int(product.price),
+                    'stock'         : product.stock,
+                    'is_new'        : product.is_new,
+                    'url'           : list(images_url),
+                    'descriptions'  : list(descriptions),
+                    'star'          : uniform(0.0,5.0),
+                    'breadcrumb'    : [
+                        product.sub_category.category.korean_name, 
+                        {'id':product.sub_category.id, 
+                        'name':product.sub_category.korean_name}, 
+                        product.korean_name]
+                    }]
```

Comment:
```suggestion
                result = 
                    {
                    'id'            : product.id,
                    'korean_name'   : product.korean_name,
                    'english_name'  : product.english_name,
                    'price'         : int(product.price),
                    'stock'         : product.stock,
                    'is_new'        : product.is_new,
                    'url'           : [image.url for image in product.image.all()],
                    'descriptions'  : list(descriptions),
                    'star'          : uniform(0.0,5.0),
                    'breadcrumb'    : [
                        product.sub_category.category.korean_name, 
                        {'id':product.sub_category.id, 
                        'name':product.sub_category.korean_name}, 
                        product.korean_name]
                    }
```
- 제품 정보 하나만 보내기 때문에 list에 안 담고 바로 딕셔러리로 보내면 될 것 같습니다.
- image url 가져오는 방법 수정해 봤는데, 참고해 보세요.
- breadcrumb의 내용과 데이터를 보내는 이유를 코멘트로 남겨주세요.

Reply:
1. 프론트 쪽에서 json을 리스트로 받아야 매핑이 가능하다고하여 리스트로 보냈습니다
2. 잘못된 코드였던 것 같습니다 수정하겠습니다
3. 이것도 비슷한 이유인데 breadcrumb를 구현하고 싶어서 리스트로 받아 인덱싱 할 수 있도록 재구성하여 보냈습니다

> ![image](https://user-images.githubusercontent.com/74296888/118798917-24381000-b8d9-11eb-894b-cf4f49a8744f.png)